adapter/internal/oasparser/model/async_api.go

```diff
@@ -146,12 +140,10 @@ func (asyncAPI AsyncAPI) getResources() []*Resource {
 			continue
 		}
 
-		if found {
-			pubOrSubVendorExtensions["x-wso2-disable-security"] = val
-		}
+		pubOrSubVendorExtensions[xWso2DisableSecurity] = val
 
 		if channelItem.XAuthType == "None" || channelItem.XWso2DisableSecurity {
-			pubOrSubVendorExtensions["x-wso2-disable-security"] = true
+			pubOrSubVendorExtensions[xWso2DisableSecurity] = true
 		}
```

Comment:
Here if API Level and resource level both security disable parameter is set, the resource level parameter will take the priority. Is that what is intended? If we set disable security at top level (API Level), user will expect it will effect all the resources right. Also, in our current UX we don't allow setting both at the same time too. What do we have in the related OpenAPI code?

Reply:
Ok, I think this behaviour is correct. We are overriding when it's set to true only. So when the resource level config overrides, it will always be disable security set to true. When the top level config also has disable security set to true, this has no effect (both are true). If the top level config doesn't have disable security, the resource level config should actually override and set.